app/controllers/alchemy/admin/languages_controller.rb

```diff
@@ -1,11 +1,28 @@
 module Alchemy
   module Admin
     class LanguagesController < ResourcesController
+      def index
+        @query = Language.on_current_site.ransack(params[:q])
+        @languages = @query.result.page(params[:page] || 1).per(per_page_value_for_screen_size)
+      end
+
       def new
         @language = Language.new
         @language.page_layout = configured_page_layout || @language.page_layout
       end
 
+      def create
+        @language = Language.new(resource_params)
+        @language.site = Site.current
+        @language.save
+
+        render_errors_or_redirect(
+          @language,
+          alchemy.admin_languages_path,
+          flash_notice_for_resource_action
+        )
+      end
```

Comment:
Oh, yes. This would actually work and we wouldn't need to override this action. Let me change this